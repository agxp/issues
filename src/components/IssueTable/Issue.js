import React, {Component} from 'react';
import Icon from './Icon';
import moment from 'moment';


export default class Issue extends Component {
  constructor(props) {
    super(props);

    this.state = {issue: null}
  }

  componentDidMount() {
    this.GetIssue();
  }

  GetIssue() {
    let issue = this.props.issue;
     if (issue) {
       console.log(issue);
       this.setState({issue})
     }
  }

  render() {
    const issue = this.state.issue;
    const openclose = this.props.open ? "opened" : "was closed";


    if (issue != null) {
        const hoverdate = moment(issue.created_at).format('MMMM Do, YYYY, hh:mm a');
        const currdate = moment().subtract(30, 'days');
        const date = moment(issue.created_at).isBefore(currdate) ? " on " + moment(issue.created_at).format('MMMM Do, YYYY') : moment(issue.created_at).fromNow();
        const assignee = issue.assignee ? <img src={issue.assignee.avatar_url} className="float-right" width={20} height={20}/> : null;

        const labels = issue.labels.map(v => {
           return <span style={{backgroundColor: "#" + v.color, padding: 5, color: "white"}} key={v.name}>{v.name}</span>
        });

        return (
          <div>
              <div className={"float-left"} style={{padding: 10}}>
                  <Icon open={this.props.open} />
              </div>

            <div style={{fontSize: 16, fontWeight: 600}}>

                <a href={issue.html_url} style={{paddingRight: 10}}>
                    {issue.title}
                </a>

                {labels}

                <a href={issue.html_url}>
                    <div className="float-right">
                        <svg viewBox="0 0 16 16" version="1.1" width="16"
                             height="16" aria-hidden="true">
                            <path d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"></path>
                        </svg>
                        {issue.comments}
                    </div>
                </a>

                {assignee}

            </div>

            <div style={{fontSize: 12}} title={hoverdate}>
                #{issue.number} {openclose} {date} by {issue.user.login}
            </div>

          </div>

      );
    }
    return null;

  }
}
