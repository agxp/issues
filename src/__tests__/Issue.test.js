import React from 'react';
import Issue from '../components/IssueTable/Issue';
import Icon from '../components/IssueTable/Icon';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

const testIssue =
    {
        "url": "https://api.github.com/repos/cloudflare/cf-ui/issues/21",
        "repository_url": "https://api.github.com/repos/cloudflare/cf-ui",
        "labels_url": "https://api.github.com/repos/cloudflare/cf-ui/issues/21/labels{/name}",
        "comments_url": "https://api.github.com/repos/cloudflare/cf-ui/issues/21/comments",
        "events_url": "https://api.github.com/repos/cloudflare/cf-ui/issues/21/events",
        "html_url": "https://github.com/cloudflare/cf-ui/issues/21",
        "id": 192062081,
        "node_id": "MDU6SXNzdWUxOTIwNjIwODE=",
        "number": 21,
        "title": "Upgrade eslint to 3+",
        "user": {
            "login": "wyuenho",
            "id": 160028,
            "node_id": "MDQ6VXNlcjE2MDAyOA==",
            "avatar_url": "https://avatars1.githubusercontent.com/u/160028?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/wyuenho",
            "html_url": "https://github.com/wyuenho",
            "followers_url": "https://api.github.com/users/wyuenho/followers",
            "following_url": "https://api.github.com/users/wyuenho/following{/other_user}",
            "gists_url": "https://api.github.com/users/wyuenho/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/wyuenho/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/wyuenho/subscriptions",
            "organizations_url": "https://api.github.com/users/wyuenho/orgs",
            "repos_url": "https://api.github.com/users/wyuenho/repos",
            "events_url": "https://api.github.com/users/wyuenho/events{/privacy}",
            "received_events_url": "https://api.github.com/users/wyuenho/received_events",
            "type": "User",
            "site_admin": false
        },
        "labels": [
            {
                "id": 390162616,
                "node_id": "MDU6TGFiZWwzOTAxNjI2MTY=",
                "url": "https://api.github.com/repos/cloudflare/cf-ui/labels/enhancement",
                "name": "enhancement",
                "color": "84b6eb",
                "default": true
            }
        ],
        "state": "closed",
        "locked": false,
        "assignee": null,
        "assignees": [],
        "milestone": null,
        "comments": 0,
        "created_at": "2016-11-28T17:09:46Z",
        "updated_at": "2016-11-30T11:43:35Z",
        "closed_at": "2016-11-30T11:43:35Z",
        "author_association": "CONTRIBUTOR",
        "body": ""
    };

test('Issue should render correctly', () => {
    const issue = shallow(<Issue issue={testIssue} open={true}/>);

    expect(issue).toMatchSnapshot();
});

test('Issue should render an Icon', () => {
    const issue = shallow(<Issue issue={testIssue} open={true}/>);

    expect(issue.containsMatchingElement(<Icon open={true}/>)).toBe(true);
});