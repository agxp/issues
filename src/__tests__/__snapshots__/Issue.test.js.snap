// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Issue
    issue={
        Object {
            "assignee": null,
            "assignees": Array [],
            "author_association": "CONTRIBUTOR",
            "body": "",
            "closed_at": "2016-11-30T11:43:35Z",
            "comments": 0,
            "comments_url": "https://api.github.com/repos/cloudflare/cf-ui/issues/21/comments",
            "created_at": "2016-11-28T17:09:46Z",
            "events_url": "https://api.github.com/repos/cloudflare/cf-ui/issues/21/events",
            "html_url": "https://github.com/cloudflare/cf-ui/issues/21",
            "id": 192062081,
            "labels": Array [
              Object {
                "color": "84b6eb",
                "default": true,
                "id": 390162616,
                "name": "enhancement",
                "node_id": "MDU6TGFiZWwzOTAxNjI2MTY=",
                "url": "https://api.github.com/repos/cloudflare/cf-ui/labels/enhancement",
              },
            ],
            "labels_url": "https://api.github.com/repos/cloudflare/cf-ui/issues/21/labels{/name}",
            "locked": false,
            "milestone": null,
            "node_id": "MDU6SXNzdWUxOTIwNjIwODE=",
            "number": 21,
            "repository_url": "https://api.github.com/repos/cloudflare/cf-ui",
            "state": "closed",
            "title": "Upgrade eslint to 3+",
            "updated_at": "2016-11-30T11:43:35Z",
            "url": "https://api.github.com/repos/cloudflare/cf-ui/issues/21",
            "user": Object {
              "avatar_url": "https://avatars1.githubusercontent.com/u/160028?v=4",
              "events_url": "https://api.github.com/users/wyuenho/events{/privacy}",
              "followers_url": "https://api.github.com/users/wyuenho/followers",
              "following_url": "https://api.github.com/users/wyuenho/following{/other_user}",
              "gists_url": "https://api.github.com/users/wyuenho/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/wyuenho",
              "id": 160028,
              "login": "wyuenho",
              "node_id": "MDQ6VXNlcjE2MDAyOA==",
              "organizations_url": "https://api.github.com/users/wyuenho/orgs",
              "received_events_url": "https://api.github.com/users/wyuenho/received_events",
              "repos_url": "https://api.github.com/users/wyuenho/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/wyuenho/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/wyuenho/subscriptions",
              "type": "User",
              "url": "https://api.github.com/users/wyuenho",
            },
          }
    }
    open={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="float-left"
          style={
                    Object {
                              "padding": 10,
                            }
          }
>
          <Icon
                    open={true}
          />
</div>,
        <div
          style={
                    Object {
                              "fontSize": 16,
                              "fontWeight": 600,
                            }
          }
>
          <a
                    href="https://github.com/cloudflare/cf-ui/issues/21"
                    style={
                              Object {
                                        "paddingRight": 10,
                                      }
                    }
          >
                    Upgrade eslint to 3+
          </a>
          <span
                    style={
                              Object {
                                        "backgroundColor": "#84b6eb",
                                        "color": "white",
                                        "padding": 5,
                                      }
                    }
          >
                    enhancement
          </span>
          <a
                    href="https://github.com/cloudflare/cf-ui/issues/21"
          >
                    <div
                              className="float-right"
                    >
                              <svg
                                        aria-hidden="true"
                                        height="16"
                                        version="1.1"
                                        viewBox="0 0 16 16"
                                        width="16"
                              >
                                        <path
                                                  d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
                                        />
                              </svg>
                              0
                    </div>
          </a>
</div>,
        <div
          style={
                    Object {
                              "fontSize": 12,
                            }
          }
          title="November 28th, 2016, 09:09 am"
>
          #
          21
           
          opened
           
           on November 28th, 2016
           by 
          wyuenho
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Icon
            open={true}
/>,
          "className": "float-left",
          "style": Object {
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "open": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href="https://github.com/cloudflare/cf-ui/issues/21"
              style={
                            Object {
                                          "paddingRight": 10,
                                        }
              }
>
              Upgrade eslint to 3+
</a>,
            Array [
              <span
                style={
                                Object {
                                                "backgroundColor": "#84b6eb",
                                                "color": "white",
                                                "padding": 5,
                                              }
                }
>
                enhancement
</span>,
            ],
            <a
              href="https://github.com/cloudflare/cf-ui/issues/21"
>
              <div
                            className="float-right"
              >
                            <svg
                                          aria-hidden="true"
                                          height="16"
                                          version="1.1"
                                          viewBox="0 0 16 16"
                                          width="16"
                            >
                                          <path
                                                        d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
                                          />
                            </svg>
                            0
              </div>
</a>,
            null,
          ],
          "style": Object {
            "fontSize": 16,
            "fontWeight": 600,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Upgrade eslint to 3+",
              "href": "https://github.com/cloudflare/cf-ui/issues/21",
              "style": Object {
                "paddingRight": 10,
              },
            },
            "ref": null,
            "rendered": "Upgrade eslint to 3+",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": "enhancement",
            "nodeType": "host",
            "props": Object {
              "children": "enhancement",
              "style": Object {
                "backgroundColor": "#84b6eb",
                "color": "white",
                "padding": 5,
              },
            },
            "ref": null,
            "rendered": "enhancement",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="float-right"
>
                <svg
                                aria-hidden="true"
                                height="16"
                                version="1.1"
                                viewBox="0 0 16 16"
                                width="16"
                >
                                <path
                                                d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
                                />
                </svg>
                0
</div>,
              "href": "https://github.com/cloudflare/cf-ui/issues/21",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <svg
                    aria-hidden="true"
                    height="16"
                    version="1.1"
                    viewBox="0 0 16 16"
                    width="16"
>
                    <path
                                        d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
                    />
</svg>,
                  0,
                ],
                "className": "float-right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "children": <path
                      d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
/>,
                    "height": "16",
                    "version": "1.1",
                    "viewBox": "0 0 16 16",
                    "width": "16",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "d": "M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "path",
                  },
                  "type": "svg",
                },
                0,
              ],
              "type": "div",
            },
            "type": "a",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "#",
            21,
            " ",
            "opened",
            " ",
            " on November 28th, 2016",
            " by ",
            "wyuenho",
          ],
          "style": Object {
            "fontSize": 12,
          },
          "title": "November 28th, 2016, 09:09 am",
        },
        "ref": null,
        "rendered": Array [
          "#",
          21,
          " ",
          "opened",
          " ",
          " on November 28th, 2016",
          " by ",
          "wyuenho",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="float-left"
            style={
                        Object {
                                    "padding": 10,
                                  }
            }
>
            <Icon
                        open={true}
            />
</div>,
          <div
            style={
                        Object {
                                    "fontSize": 16,
                                    "fontWeight": 600,
                                  }
            }
>
            <a
                        href="https://github.com/cloudflare/cf-ui/issues/21"
                        style={
                                    Object {
                                                "paddingRight": 10,
                                              }
                        }
            >
                        Upgrade eslint to 3+
            </a>
            <span
                        style={
                                    Object {
                                                "backgroundColor": "#84b6eb",
                                                "color": "white",
                                                "padding": 5,
                                              }
                        }
            >
                        enhancement
            </span>
            <a
                        href="https://github.com/cloudflare/cf-ui/issues/21"
            >
                        <div
                                    className="float-right"
                        >
                                    <svg
                                                aria-hidden="true"
                                                height="16"
                                                version="1.1"
                                                viewBox="0 0 16 16"
                                                width="16"
                                    >
                                                <path
                                                            d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
                                                />
                                    </svg>
                                    0
                        </div>
            </a>
</div>,
          <div
            style={
                        Object {
                                    "fontSize": 12,
                                  }
            }
            title="November 28th, 2016, 09:09 am"
>
            #
            21
             
            opened
             
             on November 28th, 2016
             by 
            wyuenho
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon
              open={true}
/>,
            "className": "float-left",
            "style": Object {
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "open": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="https://github.com/cloudflare/cf-ui/issues/21"
                style={
                                Object {
                                                "paddingRight": 10,
                                              }
                }
>
                Upgrade eslint to 3+
</a>,
              Array [
                <span
                  style={
                                    Object {
                                                      "backgroundColor": "#84b6eb",
                                                      "color": "white",
                                                      "padding": 5,
                                                    }
                  }
>
                  enhancement
</span>,
              ],
              <a
                href="https://github.com/cloudflare/cf-ui/issues/21"
>
                <div
                                className="float-right"
                >
                                <svg
                                                aria-hidden="true"
                                                height="16"
                                                version="1.1"
                                                viewBox="0 0 16 16"
                                                width="16"
                                >
                                                <path
                                                                d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
                                                />
                                </svg>
                                0
                </div>
</a>,
              null,
            ],
            "style": Object {
              "fontSize": 16,
              "fontWeight": 600,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Upgrade eslint to 3+",
                "href": "https://github.com/cloudflare/cf-ui/issues/21",
                "style": Object {
                  "paddingRight": 10,
                },
              },
              "ref": null,
              "rendered": "Upgrade eslint to 3+",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": "enhancement",
              "nodeType": "host",
              "props": Object {
                "children": "enhancement",
                "style": Object {
                  "backgroundColor": "#84b6eb",
                  "color": "white",
                  "padding": 5,
                },
              },
              "ref": null,
              "rendered": "enhancement",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="float-right"
>
                  <svg
                                    aria-hidden="true"
                                    height="16"
                                    version="1.1"
                                    viewBox="0 0 16 16"
                                    width="16"
                  >
                                    <path
                                                      d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
                                    />
                  </svg>
                  0
</div>,
                "href": "https://github.com/cloudflare/cf-ui/issues/21",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <svg
                      aria-hidden="true"
                      height="16"
                      version="1.1"
                      viewBox="0 0 16 16"
                      width="16"
>
                      <path
                                            d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
                      />
</svg>,
                    0,
                  ],
                  "className": "float-right",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "children": <path
                        d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
/>,
                      "height": "16",
                      "version": "1.1",
                      "viewBox": "0 0 16 16",
                      "width": "16",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "d": "M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "path",
                    },
                    "type": "svg",
                  },
                  0,
                ],
                "type": "div",
              },
              "type": "a",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "#",
              21,
              " ",
              "opened",
              " ",
              " on November 28th, 2016",
              " by ",
              "wyuenho",
            ],
            "style": Object {
              "fontSize": 12,
            },
            "title": "November 28th, 2016, 09:09 am",
          },
          "ref": null,
          "rendered": Array [
            "#",
            21,
            " ",
            "opened",
            " ",
            " on November 28th, 2016",
            " by ",
            "wyuenho",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
